### PR TITLE
DOC: stats.FitResult.plot: add example

### LINKED ISSUE #18944: How to change the default setup for using res.plot()
### Is your feature request related to a problem? Please describe.

Hi all,

I am fitting my data with a poisson distribution, and I use the commond res.plot to visualize my result. 

How can I change the line color or hist color for the default setting up?

for example, I want set the color of fitted PMF as red want to increase the linewidth.

image

Hope can get your suggestion. Thanks

### Describe the solution you'd like.

_No response_

### Describe alternatives you've considered.

_No response_

### Additional context (e.g. screenshots, GIFs)

_No response_

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
Closes gh-18944

#### What does this implement/fix?
This adds an example of editing the plot created by the `FitResult.plot` method.

#### Additional information
Fingers crossed that intersphinx will work.

### CODE REVIEW

File: scipy/stats/_fit.py [hunk 1]
```diff
@@ -143,8 +143,27 @@ def plot(self, ax=None, *, plot_type="hist"):
 
         Returns
         -------
-        ax : matplotlib.axes.Axes
+        ax : `matplotlib.axes.Axes`
             The matplotlib Axes object on which the plot was drawn.
+
+        Example
+        -------
```
Comment (mdhaber) [inline scipy/stats/_fit.py:150]:
```suggestion
        Examples
        --------
```

File: scipy/stats/_fit.py [hunk 2]
```diff
@@ -143,8 +143,27 @@ def plot(self, ax=None, *, plot_type="hist"):
...
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from scipy import stats
+        >>> import matplotlib.pyplot as plt  # matplotlib must be installed
+        >>> rng = np.random.default_rng()
+        >>> data = stats.nbinom(5, 0.5).rvs(size=1000, random_state=rng)
+        >>> bounds = [(0, 30), (0, 1)]
+        >>> res = stats.fit(stats.nbinom, data, bounds)
+        >>> ax = res.plot()  # save Matplotlib Axes object
```
Comment (mdhaber) [inline scipy/stats/_fit.py:158]:
```suggestion
        >>> ax = res.plot()  # save matplotlib Axes object
```

File: scipy/stats/_fit.py [hunk 3]
```diff
@@ -112,11 +112,11 @@ def nllf(self, params=None, data=None):
     def plot(self, ax=None, *, plot_type="hist"):
         """Visually compare the data against the fitted distribution.
 
-        Available only if ``matplotlib`` is installed.
+        Available only if `matplotlib` is installed.
 
         Parameters
         ----------
-        ax : matplotlib.axes.Axes
+        ax : `matplotlib.axes.Axes`
```
Comment (tupui) [inline scipy/stats/_fit.py:119]: Do you know why we need to do that? On another project I don't have that and it does the linking properly.

**Review by tupui — APPROVED** [review]
Thanks Matt, LGTM and renders properly.

In general I don't think this was really a valid report. If we have more, maybe we need to tweak the doc on what is `ax` (have a standard doc for `ax` with links or point to a recipe.)

Here it's ok to have an example as there was nothing.

File: scipy/stats/_fit.py (hunk 3 above)
Comment (mdhaber) [reply scipy/stats/_fit.py:119]: I don't know, but it was not rendering properly before.
https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats._result_classes.FitResult.plot.html#scipy.stats._result_classes.FitResult.plot

**Pull request merged**